charts/azurepoc/templates/deployment.yaml

```diff
@@ -0,0 +1,77 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ .Values.name }}-deployment
+  labels:
+{{ include "azurepoc.labels" . | indent 4 }}
+spec:
+  strategy:
+    # Ensures that whenever the k8s pods for this application
+    # are restarted it is done in a rolling fashion. Only 1 application instance will 
+    # be unavailable at a given time. If any of the restarts fail for any reasons the
+    # rollout is automatically halted.
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 0
+      maxUnavailable: 1
+  # Ensures that any rollback operations must be done via git in our helmfile repo
+  # rather than using kubectl rollback undo which could result in potential mismatch
+  # between an application and it's configuration
```

Comment:
I think this is great. However, I think it’s overkill.  Basic Kubernetes definitions/standards don’t need comments. If you google [revisioinHistoryLimit](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#clean-up-policy) it's clear what it's used for. Also, we're developing a linter that will enforce these standards.

Reply:
@knapii-developments I intentionally added some overkill comments here because I thought this particular chart would be good to use as a teaching tool for folks on our team learning k8s, by explaining some of the fields that you don't see often in a lot of kubernetes tutorials. And so that you don't have to constantly switch back and forth between the code and the k8s docs when you are seeing this stuff for the first time.

I agree that not every chart needs a comment explaining this field, however I think it is valuable for folks on our team to understand why we set this value the way we do as it is specific to our kubernetes implementation.

Reply:
✅